modules/@apostrophecms/ui/ui/apos/utils/index.js

```diff
@@ -1,9 +1,15 @@
 export const debounce = (func, timeout) => {
   let timer;
   return (...args) => {
-    clearTimeout(timer);
-    timer = setTimeout(() => {
-      func.apply(this, args);
-    }, timeout);
+    return new Promise((resolve, reject) => {
+      clearTimeout(timer);
+      timer = setTimeout(() => {
+        try {
+          resolve(func.apply(this, args));
+        } catch (err) {
+          reject(err);
+        }
+      }, timeout);
+    });
```

Comment:
@boutell Also made this modification, it allows to await a debounced function.
It works for an async function and will await it to be resolved.